### ISSUE TITLE
API keys do not work with graphql

### ISSUE THREAD

**Issue by furqan-shakoor** [Requarks/wiki#2426]
**Describe the bug**
We had some API keys we were using to query graphql. The keys stopped working in 2.5. They either raise a "Forbidden" error or the query times out. I'm not sure if its related but we have a keycloak integration for Authentication. Creating a local username+password, using that to login with graphql and subsequently using the jwt token works.

I've tried creating keys with "Full Access" and I've tried keys tied to Group permissions. None of those work.

**To Reproduce**
1. Go to `/a/api`
2. Create an API key with "Full Access"
3. Go to `/graphql`
4. Query anything with header "Authorization": "Bearer <API_KEY>"
5. The query either runs forever or raises a "Forbidden" error

When we get a forbidden error, these are the error details

```
{'errors': [{'message': 'Forbidden', 'locations': [{'line': 3, 'column': 3}], 'path': ['pages', 'list'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Forbidden', '    at field.resolve (/wiki/server/graph/directives/auth.js:47:17)', '    at field.resolve (/wiki/node_modules/graphql-extensions/dist/index.js:134:26)', '    at field.resolve (/wiki/node_modules/apollo-server-core/dist/utils/schemaInstrumentation.js:52:26)', '    at resolveFieldValueOrError (/wiki/node_modules/graphql/execution/execute.js:502:18)', '    at resolveField (/wiki/node_modules/graphql/execution/execute.js:460:16)', '    at executeFields (/wiki/node_modules/graphql/execution/execute.js:297:18)', '    at collectAndExecuteSubfields (/wiki/node_modules/graphql/execution/execute.js:748:10)', '    at completeObjectValue (/wiki/node_modules/graphql/execution/execute.js:738:10)', '    at completeValue (/wiki/node_modules/graphql/execution/execute.js:626:12)', '    at /wiki/node_modules/graphql/execution/execute.js:527:16', '    at runMicrotasks (<anonymous>)', '    at processTicksAndRejections (internal/process/task_queues.js:93:5)', '    at async Promise.all (index 0)']}}}], 'data': {'pages': None}}
```

**Expected behavior**
The graphql API responds

**Host Info (please complete the following information):**
 - OS: Docker Container (Linux)
 - Wiki.js version: 2.5.132
 - Database engine: postgres 11.7

**Comment by niekcandaele** [comment]
I see the same thing on 2.5.126

Maybe this helps, I see the message `warn: user.groups is not iterable` in my logs

**Comment by greenitthe** [comment]
Similar on 2.5.132. Getting `Unauthorized` message with `INTERNAL_SERVER_ERROR` code. Logs show `warn: Cannot read property 'groups' of undefined` using a full access key, or `warn: user.groups is not iterable` with group permissions.

**Comment by Igorgro** [comment]
+1, having same issue

**Comment by greenitthe** [comment]
Issue seems to start at `server/core/auth.js` line 146 during token revalidation/renewal. It seems `jwtPayload.id` is undefined in my case, which causes an error at `server/models/user.js` when the `refreshToken(user)` function attempts to access `user.groups` in`_.isNil()`

`jwtPayload` in `auth.js` appears to extract the following keys: `api, grp, iat, exp, aud, iss`. There does not appear to be an id property in the object. Unfortunately I'm not familiar enough with these tools to probe much further.

**Comment by NGPixel** [comment]
Fixed by 1404d6343ec82bb77843b5ea9b4b0d1b3c230321
